WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -532,6 +535,7 @@ def update
     @competition = competition_from_params(includes: CHECK_SCHEDULE_ASSOCIATIONS)
     @competition_admin_view = params.key?(:competition_admin_view) && current_user.can_admin_competitions?
     @competition_organizer_view = !@competition_admin_view
+    @actually_using_organizer_view = !params.key?(:competition_admin_view) # If the edit is done through this view
```

Comment:
Why is this relevant? Can't you use `@competition_organizer_view` directly above?
From what I can tell just by looking at the code itself, this creates an edge case only when somebody who isn't an admin uses the admin view, which shouldn't even be possible in the first place.

Reply:
Using `@competition_organizer_view` would result in the short name / id not automatically updating when a competition admin updates their competition name from the organiser view.

Reply:
I don't think so: If a competition admin updates their own competition from the organizer view, then...
- `params.key?(:competition_admin_view)` is `false` because they are using the **organizer** view
- `current_user.can_admin_competitions?` is `true` because they are a competition admin
- `@competition_admin_view` evaluates to `false && true` which is clearly `false`
- `@competition_organizer_view` evaluates to `!false` which is `true`.

So even in the scenario that you described, the `@competition_organizer_view` variable should be `true`, and the update trigger should run. Otherwise, I have a flaw in my logic (which you're free to point out) or it's a bug in the code

Reply:
Everything still works after your change, so I think I just misunderstood the original code those months back (your logic makes sense to me).